### PR TITLE
Feature/issue 18 - HomeScreen,More Screen - Update UI

### LINKED ISSUE #22: [Feature] HomeScreen , More Screen/ Update UI
### Overview
- // Update UI

### Impact
-  Home Screen
-  More Screen

### Screenshots (Optional)
- // Capture affected screen
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 23
s
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 33
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 45
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 49
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 58

### PR DESCRIPTION
**Pull request by minh5896**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
-  Update UI

### Solution
-  Update UI

### Impact
-  Home Screen
- More Screen

### Screenshots (Optional)
- // Home Screen
 Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 23
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 33
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 45
Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 49
- More Screen

Simulator Screen Shot - iPhone 12 Pro - 2021-11-01 at 09 07 58

### CODE REVIEW

File: Modules/Model/Model/Classes/Movie/IMovieAPIService.swift [hunk 1]
```diff
@@ -9,4 +9,6 @@ import Foundation
 
 public protocol IMovieAPIService {
 	func getMovies(completion: @escaping (Result<[IMovieModel], Error>) -> Void)
+	func getDownloadMovies(completion: @escaping (Result<[IMovieModel], Error>) -> Void)
+
```
Comment (namnh92) [inline Modules/Model/Model/Classes/Movie/IMovieAPIService.swift]:
```suggestion
```

File: MovieApp/Screens/Download/DownloadViewController.swift [hunk 2]
```diff
@@ -31,20 +31,19 @@ class DownloadViewController: BaseViewController {
 		downloadCollectionView.delegate = self
 		downloadCollectionView.dataSource = self
 		setupUI()
+		setupTableView()
```
Comment (namnh92) [inline MovieApp/Screens/Download/DownloadViewController.swift]: Move to setupUI

File: MovieApp/Screens/Download/Views/DownloadCell.swift [hunk 3]
```diff
@@ -11,15 +11,14 @@ class DownloadCell: UICollectionViewCell {
...
-			self.layer.masksToBounds = true
-			self.layer.cornerRadius = 5.0
-			self.layer.borderWidth = 2.0
-			self.layer.borderColor = UIColor.white.cgColor
+// MARK: - LifeCycle
+	override func awakeFromNib() {
+		super.awakeFromNib()
+		self.layer.masksToBounds = true
+		self.layer.cornerRadius = 5.0
+		self.layer.borderWidth = 2.0
```
Comment (namnh92) [inline MovieApp/Screens/Download/Views/DownloadCell.swift]:
```suggestion
		self.layer.cornerRadius = Constants.cornerRadius
		self.layer.borderWidth = Constants.borderWidth
```
Make constants

File: MovieApp/Screens/Home/HomeViewController.swift [hunk 4]
```diff
@@ -58,48 +67,68 @@ extension HomeViewController: IHomeViewController {
...
+			return cell
+		case .romance(let identifer, _):
+			guard let cell = self.tableView.dequeueReusableCell(withIdentifier: identifer) as? ListMovieCell else { return UITableViewCell() }
+			cell.configure(with: viewModel?.movieViewModels ?? [])
+			cell.delegate = self
+			return cell
+		case .tvShow(let identifer, _):
+			guard let cell = self.tableView.dequeueReusableCell(withIdentifier: identifer) as? TvShowViewCell else { return UITableViewCell() }
+			cell.configure(with: viewModel?.movieViewModels.randomElement())
+//			cell.delegate = self
```
Comment (namnh92) [inline MovieApp/Screens/Home/HomeViewController.swift]:
```suggestion
```
Remove comment

File: MovieApp/Screens/Home/HomeViewController.swift [hunk 5]
```diff
@@ -111,38 +140,40 @@ extension HomeViewController: UITableViewDataSource {
 			return 130.0
 		case 2 :
 			return 220.0
+		case 7:
```
Comment (namnh92) [inline MovieApp/Screens/Home/HomeViewController.swift]: What is case 2 & 7 ? 
Please make an enum and describe about it. Don't using magic number

File: MovieApp/Screens/Home/Views/ContinueCell/ContinueItemCell/ContinueItemCell.swift [hunk 6]
```diff
@@ -13,9 +13,10 @@ class ContinueItemCell: UICollectionViewCell {
...
 	@IBOutlet weak var infoContinue: UIButton!
 	@IBOutlet weak var moreContinue: UIButton!
+	
+// MARK: - lifeCycle
 	override func awakeFromNib() {
 		super.awakeFromNib()
-		self.backgroundColor = .black
 		self.layer.masksToBounds = true
 		self.layer.cornerRadius = 5.0
 		self.layer.borderWidth = 2.0
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ContinueCell/ContinueItemCell/ContinueItemCell.swift]: Make constants
```suggestion
		self.layer.masksToBounds = true
		self.layer.cornerRadius = Constants.cornerRadius
		self.layer.borderWidth = Constants.borderWidth
```

File: MovieApp/Screens/Home/Views/PreviewCell/PreviewCell.swift [hunk 7]
```diff
@@ -7,6 +7,9 @@
 
 import UIKit
 
+private enum ConstantsCell {
+	static let previewCollectionCell = "PreviewCollectionViewCell"
+}
 protocol CellDelegate: AnyObject {
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/PreviewCell/PreviewCell.swift]:
```suggestion
}

protocol CellDelegate: AnyObject {
```

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift [hunk 8]
```diff
@@ -14,9 +14,10 @@ class TvShowViewCell: UITableViewCell {
...
-	override func awakeFromNib() {
-		super.awakeFromNib()
-		// Initialization code
-	}
+
+	// MARK: - configure
+		func configure(with cellViewModel: IMovieViewModel?) {
+			titleTvShow.text = cellViewModel?.title ?? ""
+			imageTvShow.linkDownloaded(from: cellViewModel?.backdropPath ?? "")
+		}
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift]:
```suggestion

	// MARK: - configure
	func configure(with cellViewModel: IMovieViewModel?) {
		titleTvShow.text = cellViewModel?.title ?? ""
		imageTvShow.linkDownloaded(from: cellViewModel?.backdropPath ?? "")
	}
```

File: MovieApp/Screens/More/MoreViewController.swift [hunk 9]
```diff
@@ -6,11 +6,172 @@
...
+// MARK: - Variables
+	private var viewModel: MoreViewModel?
+
+// MARK: - Life cycles
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		viewModel = MoreViewModel(viewController: self)
+		self.moreTableView.delegate = self
+		self.moreTableView.dataSource = self
+		setupTableView()
```
Comment (namnh92) [inline MovieApp/Screens/More/MoreViewController.swift]: Move to setupUI

File: MovieApp/Screens/More/MoreViewController.swift [hunk 10]
```diff
@@ -6,11 +6,172 @@
...
+
+	func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+		switch indexPath.section {
+		case 0:
+			return 200.0
+		case 1:
+			return 200.0
+		default:
+			return 50.0
+		}
```
Comment (namnh92) [inline MovieApp/Screens/More/MoreViewController.swift:128]:
```suggestion
		case 0:
			return 200.0
		case 1:
			return 200.0
		default:
			return 50.0
		}
```
Make an enum or constants describe 0, 1

File: MovieApp/Screens/More/Views/ProfileCell/ProfileTableViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+extension ProfileTableViewCell: UICollectionViewDelegateFlowLayout {
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+		return CGSize(width: collectionView.frame.width / 3.0, height: collectionView.frame.height)
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+		return 15.0
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/ProfileCell/ProfileTableViewCell.swift]:
```suggestion
		return Constants.minimumLineSpacing
```

File: MovieApp/Screens/More/Views/ProfileCell/ProfileTableViewCell.swift [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+		return 10.0
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/ProfileCell/ProfileTableViewCell.swift]:
```suggestion
		return Constants.minimumInteritemSpacing
```

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+	func sharedWhatApps()
+	func sharedAlert()
+}
+
+class SharedTableViewCell: UITableViewCell {
+
+// MARK: - Iboutlets
+	@IBOutlet private weak var labelShared1: UILabel!
+	@IBOutlet private weak var labelShared2: UILabel!
+	@IBOutlet private weak var labelShared3: UILabel!
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]:
```suggestion
	@IBOutlet private weak var shareFacebookLabel: UILabel!
```
Don't using 1,2,3... in variables

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+	@IBOutlet private weak var whatappButton: UIButton!
+	@IBOutlet private weak var facebookButton: UIButton!
+	@IBOutlet private weak var mailButton: UIButton!
+
+// MARK: - Variables
+	weak var viewModel: sharedDelegate?
+	override func awakeFromNib() {
+		super.awakeFromNib()
+		setUpUI()
+		setUpbutton()
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]:
```suggestion
```
Move to setupUI

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+		}
+		private func setUpbutton() {
+			self.copyButton.layer.masksToBounds = true
+			self.copyButton.layer.cornerRadius = 5.0
+			self.copyButton.layer.borderWidth = 2.0
+			self.copyButton.layer.borderColor = UIColor.white.cgColor
+			self.textShared.layer.masksToBounds = true
+			self.textShared.layer.cornerRadius = 5.0
+			self.textShared.layer.borderWidth = 2.0
+			self.textShared.layer.borderColor = UIColor.white.cgColor
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]: Create Constants
```suggestion
			self.copyButton.layer.cornerRadius = Constants.cornerRadius
			self.copyButton.layer.borderWidth = Constants.borderWidth
			self.copyButton.layer.borderColor = UIColor.white.cgColor
			self.textShared.layer.masksToBounds = true
			self.textShared.layer.cornerRadius = Constants.cornerRadius
			self.textShared.layer.borderWidth = Constants.borderWidth
			self.textShared.layer.borderColor = UIColor.white.cgColor
```

File: MovieApp/Screens/More/Views/StandardCell/StandardTableViewCell.swift [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+class StandardTableViewCell: UITableViewCell {
+
+// MARK: - Iboutlets
+	@IBOutlet private weak var labelStandardCell: UILabel!
+
+// MARK: - Life Cycle
+	override func awakeFromNib() {
+		super.awakeFromNib()
+}
```
Comment (namnh92) [inline MovieApp/Screens/More/Views/StandardCell/StandardTableViewCell.swift]:
```suggestion
         }
```

File: MovieApp/Screens/Tabbar/TabbarViewController.swift [hunk 17]
```diff
@@ -31,6 +31,6 @@ class TabbarViewController: UITabBarController, Storyboarded, Coordinator {
 extension TabbarViewController {
 	private func setupTabbar() {
 		tabBarController?.tabBar.itemWidth = 30.0
```
Comment (namnh92) [inline MovieApp/Screens/Tabbar/TabbarViewController.swift]:
```suggestion
		tabBarController?.tabBar.itemWidth = Constants.itemWidth
```

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: Modules/Model/Model/Classes/Movie/IMovieAPIService.swift (hunk 1 above)
Comment (minh5896) [reply Modules/Model/Model/Classes/Movie/IMovieAPIService.swift]: fix complete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 2 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: update complete

File: MovieApp/Screens/Download/Views/DownloadCell.swift (hunk 3 above)
Comment (minh5896) [reply MovieApp/Screens/Download/Views/DownloadCell.swift]: fixcomplete

File: MovieApp/Screens/Download/DownloadViewController.swift (hunk 2 above)
Comment (minh5896) [reply MovieApp/Screens/Download/DownloadViewController.swift]: fix complete

File: MovieApp/Screens/Home/HomeViewController.swift (hunk 4 above)
Comment (minh5896) [reply MovieApp/Screens/Home/HomeViewController.swift]: fix complete

File: MovieApp/Screens/Home/HomeViewController.swift (hunk 5 above)
Comment (minh5896) [reply MovieApp/Screens/Home/HomeViewController.swift]: fix complete

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift (hunk 14 above)
Comment (minh5896) [reply MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]: fix complete

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift (hunk 15 above)
Comment (minh5896) [reply MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]: fix complete

File: MovieApp/Screens/Tabbar/TabbarViewController.swift (hunk 17 above)
Comment (minh5896) [reply MovieApp/Screens/Tabbar/TabbarViewController.swift]: fix complete

File: MovieApp/Screens/More/Views/StandardCell/StandardTableViewCell.swift (hunk 16 above)
Comment (minh5896) [reply MovieApp/Screens/More/Views/StandardCell/StandardTableViewCell.swift]: fix complete

File: MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift (hunk 13 above)
Comment (minh5896) [reply MovieApp/Screens/More/Views/SharedCell/SharedTableViewCell.swift]: fix complete

File: MovieApp/Screens/More/MoreViewController.swift (hunk 9 above)
Comment (minh5896) [reply MovieApp/Screens/More/MoreViewController.swift]: fix complete

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift (hunk 8 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift]: fix complete

File: MovieApp/Screens/Home/Views/ContinueCell/ContinueItemCell/ContinueItemCell.swift (hunk 6 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/ContinueCell/ContinueItemCell/ContinueItemCell.swift]: fix  complete

File: MovieApp/Screens/Home/Views/PreviewCell/PreviewCell.swift (hunk 7 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/PreviewCell/PreviewCell.swift]: fix complete

File: MovieApp/Screens/More/MoreViewController.swift (hunk 10 above)
Comment (minh5896) [reply MovieApp/Screens/More/MoreViewController.swift:128]: fix complete

**Pull request merged**